src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -329,6 +332,9 @@ protected DigestWrapper(boolean confidential) {
         byte[] clearText = null;
         try {
             clearText = unwrapCipher.update(message, offset, len - 6);
+            if (clearText == null){
+                throw log.mechProblemDuringDecryptResultIsNull(getMechanismName()).toSaslException();
+            }
```

Comment:
Same as the previous one, the null check should probably be immediately after the try catch block.